src/components/Txt/index.tsx

```diff
@@ -48,21 +63,52 @@ export const align = style({
 	cssProperty: 'text-align',
 });
 
+const Copy = styled(Box)<{ showCopyButton: 'always' | 'hover' | undefined }>`
+	cursor: pointer;
+	visibility: ${(props) =>
+		props.showCopyButton === 'always' ? 'visible' : 'hidden'};
+	&:hover {
+		visibility: visible;
+	}
+	display: inline;
+`;
+
+const iconalign = (props: ThemedTxtProps) =>
+	props.iconalign &&
+	css`
+		display: inline-block;
+		vertical-align: bottom;
+	`;
+
 const BaseTxt = styled.div<TxtProps>`
 	${align}
 	${monospace}
 	${whitespace}
-
+	${code}
 	${caps}
 	${bold}
 	${italic}
-
 	${truncate}
+	${iconalign}
 `;
-
+// style={{ display: 'inline-block', verticalAlign: 'bottom' }}
 const Factory = (tag?: string) => {
 	return asRendition<React.FunctionComponent<TxtProps>>((props: any) => {
-		return <BaseTxt as={tag} {...props} />;
+		return props.copy ? (
+			<div>
+				<BaseTxt as={tag} {...props} iconalign />
+				<Copy
+					tooltip={{ text: 'Copied!', trigger: 'click' }}
+					onClick={() => copyToClipboard(props.copy)}
+					showCopyButton={props.showCopyMode || 'always'}
+					{...props}
+				>
+					<FontAwesomeIcon icon={faCopy} />
+				</Copy>
+			</div>
```

Comment:
Why do we need to wrap this in a div? And howcome Copy is wrapping a copy icon instead of the BaseTxt? If the user provides copy, the result should be something like `<Copy><BaseTxt /></Copy>`. Any styling updates that need to be made in order for it to work properly in any contexts (consider the ones in our UI) should be applied to one of those two components.

Reply:
@myarmolinsky IMHO this is the correct structure. Copy as it was before is just a styled component that wrap an icon, as for the style we already discussed that the icon should not have any other than the font size. (ref: https://www.flowdock.com/app/rulemotion/f-rendition/threads/tcqn5E_rvx-y9M66LSJHg57t3Ki)

Also @jmakivic what about:

```
			<div>
				<BaseTxt as={tag} {...props} iconalign />
{ props.copy && (
				<Copy
					tooltip={{ text: 'Copied!', trigger: 'click' }}
					onClick={() => copyToClipboard(props.copy)}
					showCopyButton={props.showCopyMode || 'always'}
          fontSize={props.fontSize}
				>
					<FontAwesomeIcon icon={faCopy} />
				</Copy> )
			}
			</div>
```